### PR TITLE
create-operatorroles: Auto-find policies for roles

### PR DESCRIPTION
**Pull request by vkareh**
Since the operator role policies created through 'rosa create
account-roles' are tagged with the operator metadata, we can drop the
requirement for a user-set prefix and instead automatically find the
corresponding policy.

### CODE REVIEW

File: pkg/aws/policies.go [hunk 1]
```diff
@@ -210,6 +213,52 @@ func (c *awsClient) AttachRolePolicy(roleName string, policyARN string) error {
 	return nil
 }
 
+func (c *awsClient) FindPolicyARN(operator Operator, version string) (string, error) {
+	listPoliciesOutput, err := c.iamClient.ListPolicies(&iam.ListPoliciesInput{
```
Comment (jharrington22) [inline pkg/aws/policies.go:217]: I see that we're applying a scope here but larger organizations might have > 100 policies. We might want to use `[ListPoliciesPagesWithContext](https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#IAM.ListPoliciesPagesWithContext)` instead to paginate.

Comment (jharrington22) [inline pkg/aws/policies.go:217]: I see that we're applying a scope here but larger organizations might have > 100 policies. We might want to use [ListPoliciesPagesWithContext](https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#IAM.ListPoliciesPagesWithContext) instead to paginate.

File: pkg/aws/policies.go [hunk 2]
```diff
@@ -210,6 +213,52 @@ func (c *awsClient) AttachRolePolicy(roleName string, policyARN string) error {
...
+				}
+			case tags.OpenShiftVersion:
+				isTagged = true
+				if tagValue != version {
+					skip = true
+					break
+				}
+			}
+		}
+		if isTagged && !skip {
```
Comment (ciaranRoche) [inline pkg/aws/policies.go:255]: I dont know if I am misreading the logic above, but what happens if there are multiple policies with the same tag `Key(s)`? 
Would the loop above always return `skip` as true, even if one policy is valid?

Comment (vkareh) [reply pkg/aws/policies.go:255]: No, here I want to make sure that this particular policy has tags, and that _all_ three tags (namespace, name, version) match. If any of those tags do not match, I skip it and the first `for` loop moves on to the next policy. As soon as I find a policy that has all 3 tags matching, I use it

Comment (ciaranRoche) [reply pkg/aws/policies.go:255]: Perfect, thanks for the explanation.

File: pkg/aws/policies.go (hunk 1 above)
Comment (vkareh) [reply pkg/aws/policies.go]: Done

**Review by jharrington22 — COMMENTED** [review]
```
rosa create operator-roles -c jamesh-foo-48 --mode auto -y
E: Expected a valid role prefix matching ^[\w+=,.@-]+$
```

Regex validation fails because the prefix is undefined. Should run `FindPolicyARN` immediately after the interactive mode check for a `prefix` if its blank? The the rest of the parsing would complete.

**Comment by vkareh** [comment]
Fixed it, I was validating the prefix regex even if a prefix wasn't set. This new version only checks the regex if there's a prefix

**Pull request merged**